x9k3.py

```diff
@@ -44,13 +44,15 @@ class X9K3(strm.Stream):
     X9K3 class
     """
 
-    def __init__(self, tsdata=None, show_null=False):
+    def __init__(self, tsdata=None, show_null=False, segnum=0, window=None, restarted=False):
```

Comment:
The usage of window is to permit having a continue playlist in replay option. If I am watching a live playlist, the new segments are always added at the end of the playlist, while maintainin the previous ones according to the sliding window. So, when the input file stream is finished, and the tool opens it again in replay mode, it should maintain the sliding window, to add the first segment of the input file at the end (including an EXT-X-DISCONTINUITY just before). Now, the playlist is recreated with just one segment. This breaks the playout in hls.js and native player.

I really like this tool. I think you've made an impressive job. But I feel some things are missing that are important so the tool is useful, at least for my use case. I would be very open to have a live chat with you if you want, so I can show you what I'm seeing in real time., because I´d like to contribute and make this tool even better. I'm not a python developer, so probably some of my changes break some code rules. I apologize. But, for sure, some things are not working the way the HLS players are expecting. Thanks

Reply:
I've solved it back to the previous solution of restoring the window property when replaying